FirebaseVertexAI/Tests/Unit/VertexAIAPITests.swift

```diff
@@ -32,16 +34,40 @@ final class GoogleGenerativeAITests: XCTestCase {
     let filters = [SafetySetting(harmCategory: .dangerousContent, threshold: .blockOnlyHigh)]
 
     // Permutations without optional arguments.
-    let _ = GenerativeModel(name: "gemini-1.0-pro", apiKey: "API_KEY")
-    let _ = GenerativeModel(name: "gemini-1.0-pro", apiKey: "API_KEY", safetySettings: filters)
-    let _ = GenerativeModel(name: "gemini-1.0-pro", apiKey: "API_KEY", generationConfig: config)
 
-    // All arguments passed.
-    let genAI = GenerativeModel(name: "gemini-1.0-pro",
-                                apiKey: "API_KEY",
-                                generationConfig: config, // Optional
-                                safetySettings: filters // Optional
+    // TODO: Change `genAI` to `_` when safetySettings and generationConfig are added to public API.
```

Comment:
What's preventing them from being added?

Reply:
We'll just need to check those parameters (and `requestOptions`) when determining if it's the same instance in `VertexAIComponent`:
https://github.com/firebase/firebase-ios-sdk/blob/aa474f57c99025b0851c1f0eaa0b697a360e7d15/FirebaseVertexAI/Sources/VertexAIComponent.swift#L65-L79

Note: Those parameters are currently all structs so I can't use them with `@objc` in their current form (might be able to use the `[String:Any]` trick that Morgan mentioned).